src/ray/raylet/scheduling_resources.h

```diff
@@ -22,6 +22,10 @@ enum class ResourceAvailabilityStatus : int {
   kFeasible               ///< Feasible and currently available.
 };
 
+/// Alias resource_size_t to uint64_t where resource_size_t represents the
+/// resource allocations at 1/10,000th of a unit.
+using resource_size_t = uint64_t;
```

Comment:
I think this should probably be a separate class (which has an `int` field` or even `uint8_t` since it only goes from 0 to 1000 or 10000). That will prevent accidental usage errors. It will mean that you need to add some methods for doing addition and subtraction and maybe checking if it's equal to 1.

Reply:
Actually let's use `int` instead of `uint8_t`. That way we can check that it never goes below 0 and stays in the appropriate range.

Reply:
I don't think we can set the upper bound to `1*conversion_factor_` since the `NodeManager.local_available_resources_` uses `ResourceIdSet`, which uses `ResourceSets` and does not differentiate between whole and fractional resource, and, as a result, `FractionalResourceQuantity` ends up representing resources beyond 1.